apps/web/src/hooks/useCeramic.ts

```diff
@@ -1,15 +1,43 @@
-import { ComposeClient } from "@composedb/client";
 import { EthereumWebAuth, getAccountId } from "@didtools/pkh-ethereum";
 import { InjectedConnector } from "@wagmi/core";
 import { DIDSession } from "did-session";
+import { useQuery } from "react-query";
 import { useAccount } from "wagmi";
 import { trackEvent } from "../lib/analytics";
+import { composeClient } from "../lib/compose";
 
-export function useCeramic(composeClient: ComposeClient) {
+export const LOCAL_STORAGE_KEYS = {
+  DID: "denoted.ceramic.did",
+  SIGNED_RESOURCES: "denoted.ceramic.signed-resources",
+};
+
+export function useCeramic() {
   const { address } = useAccount();
 
   const connector = new InjectedConnector();
 
+  function getIsResourcesSigned(resources: string[]) {
+    if (typeof window === "undefined") {
+      return false;
+    }
+
+    const signedResources = JSON.parse(
+      localStorage.getItem(LOCAL_STORAGE_KEYS.SIGNED_RESOURCES) ?? "[]"
+    ) as string[];
+
+    return resources.every((resource) => {
+      return signedResources.includes(resource);
```

Comment:
Hmm is this a real guarantee? What happens if the user's `LocalStorage` is corrupt?

Reply:
That's a good question! I'd like to store the session in some other external storage. But to answer your question, if local storage is corrupt then we're in big trouble since a lot of our dependencies also depend on local storage